cli/single_test.go

```diff
@@ -0,0 +1,54 @@
+package cli
+
+import (
+	"bytes"
+	"context"
+	"testing"
+
+	"github.com/ipfs/go-ipfs-cmds"
+)
+
+func TestSingle(t *testing.T) {
+	req, err := cmds.NewRequest(context.Background(), nil, nil, nil, nil, &cmds.Command{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var bufout, buferr bytes.Buffer
+
+	re, exitCh := NewResponseEmitter(&bufout, &buferr, cmds.Encoders["cli"], req)
+
+	wait := make(chan struct{})
+
+	go func() {
+		if err := cmds.EmitOnce(re, "test"); err != nil {
+			t.Fatal(err)
+		}
+
+		wait <- struct{}{}
```

Comment:
Why do we need to wait here?

Reply:
That line is supposed to unblock the wait outside the goroutine. I added that before I read the `exitCh`, but yes we don't need that anymore.